### PR TITLE
Added was_new and was_modified attributes

### PR DESCRIPTION
**Pull request by rbfmendiola**

### CODE REVIEW

File: src/AuditableBehavior.php [hunk 1]
```diff
@@ -4,12 +4,22 @@ class AuditableBehavior extends Behavior
...
+
+    public function preUpdate()
+    {
+        return "\$this->wasModified = \$this->isModified();";
     }
     
     public function postHydrate()
     {
-        return '$this->originalFieldValues = $row;';
+        return "\$this->wasNew = \$this->getWasNew(); \n \$this->wasModified = \$this->getWasModified(); \n \$this->originalFieldValues = \$row;";
```
Comment (naldz) [inline src/AuditableBehavior.php]: Can you explain this line? Why are you setting the value of $this->wasNew to a method that will return the same value? Am I missing something here? Shouldn't  this be $this->wasNew = $this->isNew().

File: src/templates/ObjectMethodGetOriginalFieldValues.php [hunk 2]
```diff
@@ -16,4 +16,23 @@ public function getOriginalFieldValues($mapped = true)
     }
 
     return $returnValues;
+}
+
+/**
+ * Return true if object was new before saving
+ */
+
+public function getWasNew()
```
Comment (naldz) [inline src/templates/ObjectMethodGetOriginalFieldValues.php]: Change method name to "wasNew" just like isNew

File: src/templates/ObjectMethodGetOriginalFieldValues.php [hunk 3]
```diff
@@ -16,4 +16,23 @@ public function getOriginalFieldValues($mapped = true)
...
+{
+    return $this->wasNew;
+}
+
+
+/**
+ * Return true if object was modified
+ */
+
+public function getWasModified()
```
Comment (naldz) [inline src/templates/ObjectMethodGetOriginalFieldValues.php]: Change method name of "getWasModified" to "wasModified" to make it consistent with "isModified".

File: tests/AuditableBehaviorTest.php [hunk 4]
```diff
@@ -77,4 +76,68 @@ public function testMappedOriginalValues()
...
+        $testObject = new TestModel();
+        // $testObject->setId(101);
+        // $testObject->setEmailAddress('logancastellano@gmail.com');
+        // $this->assertEquals(false, $testObject->getWasNew());
+        // $testObject->save();
+        // $this->assertEquals(true, $testObject->getWasNew());
+        // $testObject->setEmailAddress('reynaldocastellano@gmail.com');
+        // $this->assertEquals(true, $testObject->getWasNew());
+        // $testObject->save();
+        $this->assertEquals(false, $testObject->getWasNew());
```
Comment (naldz) [inline tests/AuditableBehaviorTest.php]: Why should "$testObject->getWasNew()" be false here?

Comment (rbfmendiola) [inline tests/AuditableBehaviorTest.php]: Object isNew = true and wasNew = false, unless saved (wasNew = true).

File: src/templates/ObjectMethodGetOriginalFieldValues.php (hunk 3 above)
Comment (rbfmendiola) [inline src/templates/ObjectMethodGetOriginalFieldValues.php]: Done.

File: src/templates/ObjectMethodGetOriginalFieldValues.php (hunk 2 above)
Comment (rbfmendiola) [inline src/templates/ObjectMethodGetOriginalFieldValues.php]: Done.

File: src/AuditableBehavior.php (hunk 1 above)
Comment (rbfmendiola) [inline src/AuditableBehavior.php]: Yes. That should be it. Updated. Thanks

File: tests/AuditableBehaviorTest.php [hunk 5]
```diff
@@ -77,4 +76,63 @@ public function testMappedOriginalValues()
         $this->assertEquals($testObject->getOriginalFieldValues(true), array('id' => 101, 'email_address' => 'reynaldocastellano@gmail.com'));
     }
     
+    public function testWasNewPropertyDeclaration()
+    {
+    	$testObject = new TestModel();
+        $refTestObject = new ReflectionObject($testObject);
+        $this->assertTrue($refTestObject->hasProperty('wasNew'), 'The property "was_new" was NOT declared in model');
```
Comment (naldz) [inline tests/AuditableBehaviorTest.php]: Change "was_new" to "wasNew"

File: tests/AuditableBehaviorTest.php [hunk 6]
```diff
@@ -77,4 +76,63 @@ public function testMappedOriginalValues()
...
+    }
+
+    public function testWasModifiedPropertyDeclaration()
+    {
+    	$testObject = new TestModel();
+        $refTestObject = new ReflectionObject($testObject);
+        $this->assertTrue($refTestObject->hasProperty('wasModified'), 'The property "was_modified" was NOT declared in model');
```
Comment (naldz) [inline tests/AuditableBehaviorTest.php]: Change "was_modified" to "wasModified"

**Pull request merged**